app/lib/pre_assembly/from_staging_location/structural_builder.rb

```diff
@@ -108,6 +110,10 @@ def administrative(fileset_file)
         { sdrPreserve: preserve, publish:, shelve: }
       end
 
+      def role?(fileset_file)
+        fileset_file.file_attributes.key? :role
```

Comment:
Have you found that we only have this key in the attrs when there's a value? It's prolly no possible to have a key and a blank value, I reckon.